### PR TITLE
Pull patch reword commit

### PR DESCRIPTION
**Pull request by derteta**
- **PR Description**
https://github.com/jesseduffield/lazygit/issues/1963

- **Please check if the PR fulfills these requirements**

* [X] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [X] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] Docs (specifically `docs/Config.md`) have been updated if necessary
* [X] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/gui/custom_patch_options_panel.go [hunk 1]
```diff
@@ -177,11 +177,20 @@ func (gui *Gui) handlePullPatchIntoNewCommit() error {
...
 
-	return gui.c.WithWaitingStatus(gui.c.Tr.RebasingStatus, func() error {
-		commitIndex := gui.getPatchCommitIndex()
-		gui.c.LogAction(gui.c.Tr.Actions.MovePatchIntoNewCommit)
-		err := gui.git.Patch.PullPatchIntoNewCommit(gui.State.Model.Commits, commitIndex)
-		return gui.helpers.MergeAndRebase.CheckMergeOrRebase(err)
+	commitIndex := gui.getPatchCommitIndex()
+	head_message, _ := gui.git.Commit.GetCommitMessage(gui.State.Model.Commits[commitIndex].FullRefName())
+
+	// TODO: use the commit message panel here
```
Comment (mark2185) [inline pkg/gui/custom_patch_options_panel.go:183]: This will probably have to wait for [this](https://github.com/jesseduffield/lazygit/pull/2390).

Comment (jesseduffield) [reply pkg/gui/custom_patch_options_panel.go:183]: That PR is merged now @derteta

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Now that we've got a new way of invoking the commit message panel, we can use that in this PR. Lemme know if you need pointers @derteta

**Comment by stefanhaller** [comment]
@derteta I took the liberty of opening a new PR that does this using the new commit message panel; hope you don't mind! See #2800.

**Comment by jesseduffield** [comment]
Looks like we can close that off in this case

**Comment by derteta** [comment]
Thanks for taking care of it, @stefanhaller !
I dropped the ball on this one :disappointed:

**Comment by jesseduffield** [comment]
No stress @derteta!

**Pull request closed without merging**